test/integration/cni/service_connectivity_test.go

```diff
@@ -126,9 +133,17 @@ var _ = Describe("[CANARY] test service connectivity", func() {
 		_, err = f.K8sResourceManagers.JobManager().
 			CreateAndWaitTillJobCompleted(negativeTesterJob)
 		Expect(err).To(HaveOccurred())
+
+		By("checking number of ENI's is less than or equal to max ENI")
```

Comment:
nit: "ENI's" -> "ENIs"
Let's add a comment here that after each test case, we validate that no more than `maxENIs` have been allocated to verify that IPs have not leaked or gotten stuck in cooldown. We choose 3 because pod placement is not guaranteed to be equally distributed.